File: java/algos/src/main/java/com/leetcode/BinaryTree.java
package com.leetcode;

import java.util.LinkedList;
import java.util.Queue;

// binary tree traversal in level order
public class BinaryTree {
    TreeNode<Integer> tree;

    public void printLevelOrder() {
        Queue<TreeNode<Integer>> queue = new LinkedList<>();
        queue.add(tree);
        while (!queue.isEmpty()) {
            TreeNode<Integer> curr = queue.poll();
            if (curr == null) {
                return;
            }
            queue.add(curr.left);
            queue.add(curr.right);
            System.out.print(curr.val);
        }
    }

    public void setTree(TreeNode<Integer> tree) {
        this.tree = tree;
    }

}

File: javascript/li-learning/binary-tree.js
class BinaryTreeNode {
  constructor(val) {
    this.left = this.left;
    this.right = this.right;
    this.val = val;
  }
}

/**
 *   a
 *  /\
 * b  c
 * /\ \
 * de  f
 *
 */

const root = new BinaryTreeNode("a");
root.left = new BinaryTreeNode("b");
root.right = new BinaryTreeNode("c");
root.left.left = new BinaryTreeNode("d");
root.left.right = new BinaryTreeNode("e");
root.right.right = new BinaryTreeNode("f");
